### PR TITLE
Improve chained and defaultazurecredential error messages

### LINKED ISSUE #8041: DefaultAzureCredential failure could be more explicit, give breadcrumbs to fix.
Currently the failure reads as follows, if no env vars are populated:

azure.core.exceptions.ClientAuthenticationError: No valid token received. EnvironmentCredential: Incomplete environment configuration.. ImdsCredential: IMDS endpoint unavailable

This was not particularly enlightening to me, and made me expect that I had broken something/specified an endpoint wrong/had some network cut.  The documentation ended up being perfectly clear ([Here](https://docs.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet-preview) was good, [Here](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/identity/azure-identity#defaultazurecredential) was even better) so perhaps we could give a breadcrumb to that if nothing else.

My naïve fix would likely include a get_token wrapper at the DefaultAzureCredential level to catch the ClientAuthenticationError and supplement the message with a brief description of the nature of the credential (chaining) and perhaps a breadcrumb to our documentation.  If this seems reasonable feel free to simply reassign the issue to me and I'll gladly implement.

To call out very clearly the Before/After for @mayurid:
**Before:**
azure.core.exceptions.ClientAuthenticationError: No valid token received. EnvironmentCredential: Incomplete environment configuration.. ImdsCredential: IMDS endpoint unavailable
**After:**
azure.core.exceptions.ClientAuthenticationError:
Chained credential request has failed, no valid token received.
Attempted credentials:
&nbsp;&nbsp;&nbsp;&nbsp;EnvironmentCredential: Incomplete environment configuration.
&nbsp;&nbsp;&nbsp;&nbsp;ImdsCredential: IMDS endpoint unavailable

Please visit the Azure identity Python SDK docs at
https://aka.ms/python-sdk-identity#defaultazurecredential
to learn what options defaultAzureCredential supports

### PR DESCRIPTION
**Pull request by KieranBrantnerMagee**
Adds slightly more detail, spacing.

Closes #8041

### CODE REVIEW

File: sdk/identity/azure-identity/azure/identity/_credentials/default.py [hunk 1]
```diff
@@ -42,3 +44,9 @@ def __init__(self, **kwargs):
             )
 
         super(DefaultAzureCredential, self).__init__(*credentials)
+
+    def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
+        try:
+            return super(DefaultAzureCredential, self).get_token(*scopes, **kwargs)
+        except ClientAuthenticationError as e:
+            raise ClientAuthenticationError(message="{}\nPlease visit the python azure identity sdk docs on github to learn what options defaultAzureCredential supports".format(e.message))
```
Comment (chlowell) [inline sdk/identity/azure-identity/azure/identity/_credentials/default.py]: @kaerm do we already have an aka link to https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/identity/azure-identity?

Comment (kaerm) [reply sdk/identity/azure-identity/azure/identity/_credentials/default.py]: how about https://aka.ms/sdk-identity ?

Comment (chlowell) [reply sdk/identity/azure-identity/azure/identity/_credentials/default.py]: Must distinguish languages; python-sdk-identity?

Comment (kaerm) [reply sdk/identity/azure-identity/azure/identity/_credentials/default.py]: you are correct - long day: https://aka.ms/python-sdk-identity

Comment (KieranBrantnerMagee) [reply sdk/identity/azure-identity/azure/identity/_credentials/default.py]: Thanks folks! (for both the aka link and sanity checking that we're OK putting URIs in exceptions) Adjusting.  :)

File: sdk/identity/azure-identity/azure/identity/_credentials/default.py [hunk 2]
```diff
@@ -42,3 +44,13 @@ def __init__(self, **kwargs):
...
+
+    def get_token(self, *scopes, **kwargs):
+        try:
+            return super(DefaultAzureCredential, self).get_token(*scopes, **kwargs)
+        except ClientAuthenticationError as e:
+            raise ClientAuthenticationError(message="""
+{}\n\nPlease visit the Azure identity Python SDK docs at 
+https://aka.ms/python-sdk-identity#defaultazurecredential 
+to learn what options DefaultAzureCredential supports"""
+                .format(e.message))
```
Comment (schaabs) [inline sdk/identity/azure-identity/azure/identity/_credentials/default.py]: @chlowell  Shouldn't this also contain the error messages from the other credentials in the chain as well?

Comment (chlowell) [reply sdk/identity/azure-identity/azure/identity/_credentials/default.py]: It does, this catches `ChainedTokenCredential`'s exception, whose message has all that, and tacks `DefaultAzureCredential` help to the end.

**Pull request merged**